ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1892,6 +1955,65 @@ void addChunkId(StoreKey storeKey, long chunkSize, int chunkIndex) {
       indexToChunkIdsAndChunkSizes.put(chunkIndex, new Pair<>(storeKey, chunkSize));
     }
 
+    /**
+     * Validate the reserved metadata chunk id is passed in correctly where it needs to be specified.
+     * @throws RouterException in case of any errors.
+     */
+    private void validateReservedMetadataChunkId() throws RouterException {
+      if (!options.isChunkUpload() && Objects.nonNull(passedInBlobProperties.getReservedMetadataBlobId())) {
```

Comment:
Just a question, right now our code set passedInBlobProperties.reservedMetadataBlobId already for the chunk upload case?

Reply:
Yes. For chunked upload when a user creates a signed url, the reserved metadata id is embedded in the signed url (AmbryUrlSigningService.java:180). And when the user attempts to upload using the signed url, the reserved metadata chunk is parsed from the signed header in `PostBlobHandler::152`.

Reply:
We will not. `isReservedMetadataEnabled` will affect only whether the reserved metadata chunk id is sent to storage nodes or not.